### PR TITLE
add metrics.md

### LINKED ISSUE #15: Establish metrics requirements

### PR DESCRIPTION
**Pull request by irrationalagent**
closes #15 
first version of the metrics plan, sure to evolve more. 

the main differences from [the ios version](https://github.com/mozilla-lockbox/lockbox-ios/blob/master/docs/metrics.md) are:

1. discussion of android-specific telemetry libraries,
2. tweaked some of the hypotheses related to analysis, given autofill will be in v1
3. added a short section on the adjust sdk

### CODE REVIEW

**Comment by devinreams** [comment]
One more pass and then product/engineering should take a look before we merge.

Then on to #14 to implement it...

File: docs/metrics.md [hunk 1]
```diff
@@ -0,0 +1,175 @@
...
+	* `category`: action
+	* `method`: show
+	* `object`: entryDetail
+	* `value`: null
+	* `extras`: ["itemid" : itemid]
+
+6. When a user taps one of the copy buttons available after being shown entry details:
+	* `category`: action
+	* `method`: tap
+	* `object`: entryCopyUsernameButton, entryCopyPasswordButton
```
Comment (linuxwolf) [inline docs/metrics.md]: Use snake case instead of camelCase (matches iOS)

File: docs/metrics.md [hunk 2]
```diff
@@ -0,0 +1,175 @@
...
+	* `category`: action
+	* `method`: show
+	* `object`: entryList
+	* `value`: null
+	* `extras`: null
+
+5. When a user shows the details of an item in the entry list:
+	* `category`: action
+	* `method`: show
+	* `object`: entryDetail
```
Comment (linuxwolf) [inline docs/metrics.md]: Use snake case instead of camelCase (matches iOS).

File: docs/metrics.md [hunk 3]
```diff
@@ -0,0 +1,175 @@
...
+	* `category`: action
+	* `method`: show
+	* `object`: login_welcome, login_fxa, login_learn_more, biometrics_setup, autofill_setup
+	* `value`: null
+	* `extras`: null
+
+4. When the main item list is shown to the user:
+	* `category`: action
+	* `method`: show
+	* `object`: entryList
```
Comment (linuxwolf) [inline docs/metrics.md]: Use snake case instead of camelCase (matches iOS).

File: docs/metrics.md [hunk 4]
```diff
@@ -0,0 +1,175 @@
...
+## List of Proposed Events
+
+1. When the app starts up:
+	* `category`: action
+	* `method`: startup
+	* `object`: app
+	* `value`: null
+	* `extras`: null
+
+2. When locking/unlocking the app:
```
Comment (linuxwolf) [inline docs/metrics.md]: do we want to distinguish between autolocking and manual locking?

File: docs/metrics.md [hunk 5]
```diff
@@ -0,0 +1,175 @@
...
+	* `category`: action
+	* `method`: tap
+	* `object`: reveal_password
+	* `value`: true or false, whether the pw is displayed
+	* `extras`: null
+
+8. When one of the settings pages is shown to the user:
+	* `category`: action
+	* `method`: show
+	* `object`: settings_list, settings_autolock, settings_preferred_browser, settings_account, settings_faq, settings_provide_feedback, settings_autofill
```
Comment (linuxwolf) [inline docs/metrics.md]: Discussed some more in IRL:
* add an event for opening the navigation drawer (`app_drawer_opened`)
* remove event for `settings_preferred_browser`

File: docs/metrics.md [hunk 6]
```diff
@@ -0,0 +1,175 @@
...
+	* `value`: null
+	* `extras`: null
+
+9. When a user changes something on the settings page:
+	* `category`: action
+	* `method`: settingsChanged
+	* `object`: settings_biometric_login, settings_autolock_time, settings_reset, settings_visual_lock, settings_preferred_browser, settings_record_usage_data
```
Comment (linuxwolf) [inline docs/metrics.md]: Discussed some more in IRL:

* remove events for `settings_preferred_browser`, `settings_record_usage_data`
* determine what `settings_visual_lock` maps to (or remove)

**Comment by irrationalagent** [comment]
@sandysage this is good for you to look over now, at your leisure.

File: docs/metrics.md [hunk 7]
```diff
@@ -0,0 +1,184 @@
...
+	* What proportion of new Lockbox users are pre-existing Firefox Account users?
+	* What proportion of users start the Account sign-in process but never complete it?
+* Does adoption of Lockbox lead to adoption of Firefox Mobile browsers (e.g. Focus)?
+	* Do users set the default browser in Lockbox to be a Firefox browser?
+
+In addition to answering the above questions that directly concern actions in the app, we will also be analyzing telemetry emitted from the password manager that exists in the the Firefox desktop browser. These analyses will primarily examine whether users of Lockbox start active curation of their credentials in the desktop browser (Lockbox users will not be able to edit credentials directly from the app).
+
+## Collection
+
+*Note: There is currently a new mozilla mobile telemetry sdk under development. Lockbox for android may end up using that sdk if it ships before the app is ready. If not, Lockbox will use the*
```
Comment (linuxwolf) [inline docs/metrics.md]: Unless the new telemetry sdk ships before the end of October 2018, I don't think we'll have time to incorporate it.  I'll add an issue for us to upgrade to this SDK once it's available.

Also, the last sentence looks unfinished.

Comment (linuxwolf) [reply docs/metrics.md]: Created #127 for updating to the new SDK when released.

File: docs/metrics.md (hunk 4 above)
Comment (sashei) [reply docs/metrics.md]: they aren't differentiated on iOS.

File: docs/metrics.md [hunk 8]
```diff
@@ -0,0 +1,184 @@
...
+		* The average rate with which a user retrieves a credential or reveals a password
+		* The distribution of above rates across all users
+* How often do users access Lockbox credentials via autofill, versus directly through the credential list provided by the app?
+* Once downloaded, do users continue to use the app? (i.e., how well are they retained?)
+	* We will count a user as retained in a given time interval if they perform one of the following actions:
+		* Display the credential list
+		* Tap a credential in the credential list
+		* Copy a credential to the clipboard
+		* Reveal a password
+		* Autofill a credential stored in Lockbox into another app
```
Comment (sandysage) [inline docs/metrics.md]: autofill into another app or another browser, right? are those tracked seperately or are browsers effectively apps?

File: docs/metrics.md [hunk 9]
```diff
@@ -0,0 +1,184 @@
...
+		* Tap the URI associated with a credential (to open it in an app or browser)
+	* Since they can be performed automatically, we will **not** count a user as retained if they *only* perform the following actions (in absence of any in the list above):
+		* Unlock their credentials
+		* Sync their credentials from the Firefox desktop browser
+* Does requiring a Firefox Account constitute a roadblock to adoption?
+	* What proportion of new Lockbox users are pre-existing Firefox Account users?
+	* What proportion of users start the Account sign-in process but never complete it?
+* Does adoption of Lockbox lead to adoption of Firefox Mobile browsers (e.g. Focus)?
+	* Do users set the default browser in Lockbox to be a Firefox browser?
```
Comment (sandysage) [inline docs/metrics.md]: We aren't expecting to have default browser selection in Lockbox as users can already set other browsers to be their default on Android.

File: docs/metrics.md (hunk 8 above)
Comment (linuxwolf) [reply docs/metrics.md]: Browsers are apps.

**Pull request merged**